mmv1/templates/terraform/update_encoder/pubsub_subscription.erb

```diff
@@ -12,6 +12,10 @@
 	# See the License for the specific language governing permissions and
 	# limitations under the License.
 -%>
-	newObj := make(map[string]interface{})
-	newObj["subscription"] = obj
+  newObj := make(map[string]interface{})
+  if d.HasChange("detached") {
```

Comment:
I believe this will cause problems if you try to both set detatched = false and update the subscription at the same time.

What action do we need to take to reattach a detatched subscription?

Reply:
Sorry for the late reply - a detached subscription can't be reattached as per the [documentation](https://cloud.google.com/pubsub/docs/create-subscription#detach_a_subscription_from_a_topic):

> After the Pub/Sub service detaches the subscription from the topic, the Pub/Sub service deletes any messages that it retains for the subscription. You can't retrieve these messages from the subscription or reattach the subscription to a topic.